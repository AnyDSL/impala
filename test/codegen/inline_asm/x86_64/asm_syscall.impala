// codegen broken

fn main() -> int {
    let str = "I'm a string, get me out of here!\n";
    let len = 34;

    // prints a string
    asm("syscall;"                      /* call kernel */
        :: "{rax}"(1), "{rdi}"(1), "{rsi}"(&str), "{rdx}"(len)
       );

    // exits with 0
    asm("syscall;"                      /* Enter kernel mode */
        :: "{rax}"(60), "{rdi}"(0)
       );

    1
}
