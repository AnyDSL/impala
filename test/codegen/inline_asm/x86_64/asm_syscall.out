I'm a string, get me out of here!
